src/lamplib/src/tests/test_preprocess.py

```diff
@@ -628,3 +628,71 @@ def test_numexpr_with_dict_string_value_throws(self):
 Test: {^NumExpr: {withExpression: "a - b", andValues: {a: *Param1, b: "25"}}}
 """
         self._assertParseException(yaml_input)
+
+    def test_flatten_array_valid_type_arrays(self):
+        yaml_input = """SchemaVersion: 2018-07-01
+Document:
+  ^FlattenArray:
+  - 1
+  - "2"
+  - [3, "4"]
+  - ^FlattenArray: [5, [6, "7", ["8", 9]]]
```

Comment:
How about something like

```
- ^FlattenArray: [5, [6, "7", ["8", 9, [], [], [1], [[[1]], [False], [""]]]
```
(I lost track of my brackets there but that's kinda the point :)